### PR TITLE
DATA-3096: Update training script for classification to take in command line args

### PR DESCRIPTION
**Pull request by vpandiarajan20**
Adds labels, epochs, and model type as CLI arguments so user can customize the values
Single Label Classification:
- Changed labels from default orange triangle and blue star to blue star and green square
- Changed num epochs to 3
Screenshot 2024-09-26 at 4 25 50 PM
Screenshot 2024-09-26 at 4 26 02 PM

> `vignesh.pandiarajan@viam.com@ROBOT-GJ3J9QVM9L rdk % go run cli/viam/main.go train submit custom with-upload --dataset-id=6515b637a420eb685e988e84 --model-org-id=0fbe951e-d4c6-427f-985f-784b7b85842c --model-name=vig-classification-C --model-type=single_label_classification --framework=tflite --path=/Users/vignesh.pandiarajan@viam.com/viam-modules-originals/cli-args-classification.tar.gz --script-name=versionCC --args=num_epochs=3,labels="'green_square blue_star'"`

Multi Label Classification:

- Changed model type from default single label to multi label
- Changed num epochs to 3
 
Screenshot 2024-09-26 at 4 40 49 PM
Screenshot 2024-09-26 at 4 41 15 PM

> `vignesh.pandiarajan@viam.com@ROBOT-GJ3J9QVM9L rdk % go run cli/viam/main.go train submit custom with-upload --dataset-id=6515b637a420eb685e988e84 --model-org-id=0fbe951e-d4c6-427f-985f-784b7b85842c --model-name=vig-classification-D --model-type=multi_label_classification --framework=tflite --path=/Users/vignesh.pandiarajan@viam.com/viam-modules-originals/cli-args-classification.tar.gz --script-name=versionDD --args=num_epochs=3,model_type=multi_label`

### CODE REVIEW

**Comment by vpandiarajan20** [comment]
Single Label Classification:
- Changed labels from default orange triangle and blue star to blue star and green square
- Changed num epochs to 225
Screenshot 2024-09-20 at 4 54 28 PM
Screenshot 2024-09-20 at 4 53 55 PM
image
Multi Label Classification:

- Changed model type from default single label to multi label
- Changed num epochs to 225
 
Screenshot 2024-09-20 at 4 21 24 PM

Screenshot 2024-09-20 at 4 20 48 PM
image

File: model/__pycache__/__init__.cpython-312.pyc
Comment (tahiyasalam) [inline model/__pycache__/__init__.cpython-312.pyc:1]: We should revert whichever commit included these cached files, otherwise they'll be part of our GH repo!

File: model/__pycache__/training.cpython-312.pyc
Comment (tahiyasalam) [inline model/__pycache__/training.cpython-312.pyc:1]: Same comment here

File: tests/__pycache__/test_file_exists.cpython-312-pytest-8.3.3.pyc
Comment (tahiyasalam) [inline tests/__pycache__/test_file_exists.cpython-312-pytest-8.3.3.pyc:1]: Same comment here about cached files

File: tests/test_label_change.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import model.training
+
+
+def test_label_change():
```
Comment (tahiyasalam) [inline tests/test_label_change.py:4]: maybe this should be test arguments or something similar since we're checking that the arguments that are passed through are correctly parsed

**Review by tahiyasalam — COMMENTED** [review]
Looks great! For these and the other PRs, can we update the PR name to include both ticket number + high-level description & add testing + more context to the comment containing the description of the PR?

File: tests/test_label_change.py (hunk 1 above)
Comment (vpandiarajan20) [reply tests/test_label_change.py:4]: shoot yea

**Comment by vpandiarajan20** [comment]
Tahiya, I'm realizing now that in order to make the application of nargs work with the command line, there will have to be changes made to app. Specifically nargs does not work when the format is `--labels="green_triangle orange_square`, it must be `--labels green_triangle orange_square`. So I might change https://github.com/vpandiarajan20/app/blob/84866071a86af43417e1b8fe2c22f7c49aaa3f8b/mltraining/vertex.go#L205
to `pythonPackageSpec.Args = append(pythonPackageSpec.Args, fmt.Sprintf("--%s %s", key, value))`.
